### ISSUE TITLE
Unable to log in after attempting to annotate (Firefox & Chrome)

### ISSUE THREAD

**Issue by klemay** [hypothesis/client#1221]
Reported by [a user in Zendesk](https://hypothesis.zendesk.com/agent/tickets/5373), I was able to replicate. 

### Steps to reproduce
1. In a browser where your login info isn't cached, go to any web page and activate Hypothesis (Chrome extension OR Bookmarklet; doesn't matter). Try http://pshannon.net/daylight/ for example. 
2. Without logging in, try to create an annotation. The sidebar will show you a message that says "You must be logged in to create annotations and highlights." 
3. Click on the "Log in" link at the top right of the sidebar

### Expected behaviour
The login window appears, you enter your credentials, click Log in, and start annotating.

### Actual behaviour
The login window never shows up. The sidebar shows an error message:
<img width="481" alt="Screen Shot 2019-07-01 at 3 53 04 PM" src="https://user-images.githubusercontent.com/12087705/60462835-6211e980-9c18-11e9-80ca-837207700792.png">

If you refresh the browser window you'll be able to log in without a problem.

### Browser/system information
The screenshot shows me going through the steps above using the Chrome extension:

Version: 1.178.0
User agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_14_5) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/75.0.3770.100 Safari/537.36

### Additional details
Same thing happens in Firefox, but the error message is slightly different: 
<img width="481" alt="Screen Shot 2019-07-01 at 3 55 43 PM" src="https://user-images.githubusercontent.com/12087705/60462991-bfa63600-9c18-11e9-998b-2569b707654d.png">

I don't run into this issue in Safari— the sign-in pop-up appears as expected. Notably, I'm able to login just fine in FF and Chrome if I run through this process in a browser window where my login info is saved. 

This doesn't seem to be urgent/critical (since a browser window refresh will allow you to log in), so I'm documenting here & will add this to the bug backlog for review.

**Comment by robertknight** [comment]
Hi @klemay - I wasn't able to reproduce this just now using the Chrome extension (v1.178.0). Could you check to see if an error appears in the browser's dev tools console when this happens, and if so, can you post that error here including the stack trace?

**Comment by klemay** [comment]
@robertknight - you may need to clear your cache before reproducing in Chrome (I wasn't able to reproduce either, until I cleared my cache & cookies). If that doesn't work, I can check browser dev tools and post results when I sign on tomorrow!

**Comment by LMS007** [comment]
I believe I saw this error several weeks ago, but I too am unable to repro this. 

- Logged out of side bar
- Logged out of https://web.hypothes.is/
- Deleted hypothesis related storage data (for extension under Application tab in dev console)
- Deleted cookies related to https://web.hypothes.is/
- Cleared local Chrome cache with hard reload. (hold down refresh icon when dev console open)
- Opened up side bar, clicked log in

_See the login popup without any errors_

@klemay Can you confirm you can still repro this? Maybe we can screen share next week?

**Comment by robertknight** [comment]
I've cleared the assignee list for this ticket as we need help with reproducing it.

**Comment by klemay** [comment]
@robertknight and @LMS007 - I'm no longer able to reproduce this, so I think we should close this.

Both FF and Chrome have released updates since I was last able to repro, so that could be a reason why we're unable to do so now.

**Comment by hmstepanek** [comment]
We are currently seeing this bug in QA: see [slack thread](https://hypothes-is.slack.com/archives/C1M8NH76X/p1573234838006100).

**Comment by LMS007** [comment]
From @hmstepanek in slack

https://hypothes-is.slack.com/archives/C1M8NH76X/p1573234838006100


```
To reproduce:
Logout of the sidebar
Try to create an annotation
Login
You should see that cannot read property 'split' of null error
```
[splitwhenloginafterannotatingbug.mov.zip](https://github.com/hypothesis/client/files/3832565/splitwhenloginafterannotatingbug.mov.zip)

**Comment by ajpeddakotla** [comment]
Dug on Sentry for this - and this came up. It seems that the frequency of this error has increased pretty significantly over the last couple of days https://sentry.io/organizations/hypothesis/issues/1324930721/events/?project=69811&statsPeriod=14d 
<img width="1036" alt="Screen Shot 2019-11-12 at 12 22 53 PM" src="https://user-images.githubusercontent.com/7463523/68699045-dace6e00-0547-11ea-944a-f7885b65012e.png">

Here's the Sentry link with all of the errors for `split null` that have been reported over the last several days: https://sentry.io/organizations/hypothesis/issues/?project=69811&query=is%3Aunresolved+split&statsPeriod=14d

**Comment by hmstepanek** [comment]
Looks like some of those are actually on PROD too.

**Comment by lyzadanger** [comment]
Hi, all. I can reliably reproduce 100% of the time by following the steps here. I have been able to reproduce this with the extension and production embedded client, but I am most concerned with being able to reproduce locally (embedded), and this will do it every time:

1. Log in as anyone
2. Select text to annotate, add some annotation body text
3. *This is the catalyst* Select "Only Me" from the publish-annotation drop-down
4. Save the annotation
5. Log out from the client (you do not need to log out of the backend/`h`)
6. Select some text to annotate and click Annotate
7. Profit/Boom

I have tracked down the crux of the issue to the (crufty/complex) `annotation` Angular controller, in a place where it tries to create a draft for this new annotation and blows up because there is no user data—ergo the permissions get populated all wonky and broken.

I'll write this up more concisely as I zero in more on the higher-level cause (versus the lower-level methods that are exploding) as the proper approach to fixing this hasn't yet made itself apparent.
